### ISSUE TITLE
X-axis doesn’t render labels consistently, not according to the number of data points

### ISSUE THREAD

**Issue by inglkruiz** [apexcharts/apexcharts.js#1751]
# Bug report

## Codepen

https://codepen.io/inglkruiz/pen/pogLagB

## Explanation

- What is the behavior you expect?
I expect to have a consistent number of x-axis' labels given the number of data points. I have the purpose of showing only the first and last x-axis' labels.
- What is happening instead?
IF I set the x-axis `format` to `dd/MM/yyyy` the 1st label disappears. 
![image](https://user-images.githubusercontent.com/12603425/86795158-9317cc00-c06d-11ea-829e-dca948b68750.png)

Let's say I set it to `dd/MM/yy` then it shows the labels accordingly but my `css` rules cannot handle all data points cases, for instance, for 1 single data point it shows dates out of the range.
![image](https://user-images.githubusercontent.com/12603425/86794984-64015a80-c06d-11ea-91c3-2eee89d3852a.png)

![image](https://user-images.githubusercontent.com/12603425/86795330-bcd0f300-c06d-11ea-9026-82c3876a31a9.png)

- What error message are you getting?
There isn’t a consistent behaviour when data points change, sometimes I get a spare `.apexcharts-xaxis-label`.

**Comment by Carniatto** [comment]
@junedchhipa any update on this bug?

**Comment by junedchhipa** [comment]
Try this

```javascript
  xaxis: {
    type: 'datetime',
    tickAmount: 1,
    labels: {
      rotateAlways: true,
      minHeight: 180,
      formatter: function(val, timestamp, opts) {
        return opts.dateFormatter(new Date(timestamp), 'dd MMM yyyy')
      }
    },
  }
```

Update Codepen - https://codepen.io/apexcharts/pen/NWrPyJY
![Screenshot 2020-10-09 at 5 55 44 PM](https://user-images.githubusercontent.com/17950663/95582730-abc2c400-0a58-11eb-8892-67cd66a313f1.png)

**Comment by inglkruiz** [comment]
@junedchhipa I played with your suggestion in codepen. 

Unfortunately, your suggestion doesn’t fix what I'm looking for because when I change to one single (1) data point
```js
// 1 data-point

var TIME = ['2019-01-01'];
var VALUES = generateValues(1);
```

the date:

1. is not aligned with the bar
2. shows dates out of the range.
![image](https://user-images.githubusercontent.com/12603425/98122517-ffd29400-1eb0-11eb-88fd-8c9358e1d3f0.png)

What I'm looking for is a consistent behaviour when data points change (including having one single DP). For instance, 1 DP -> 1 X-axis label, 2 -> 2, 3 -> 3 and so on.

I could accept any CSS hack.

PD: I did read in Docs for the X-axis:

> `Note: tickAmount doesn’t have any effect when xaxis.type = datetime`

But in fact, it does have an effect when I set `tickAmount: 'dataPoints'`, moreover, in the suggestion is written  `tickAmount: 1`. I shouldn’t according to the Docs but I validated it has an effect. I found it misleading.
